exporter/exporterhelper/internal/queuebatch/config.go

```diff
@@ -0,0 +1,104 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package queuebatch // import "go.opentelemetry.io/collector/exporter/exporterhelper/internal/queuebatch"
+
+import (
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter/exporterbatcher"
+)
+
+// Config defines configuration for queueing and batching incoming requests.
+type Config struct {
+	// Enabled indicates whether to not enqueue and batch before exporting.
+	Enabled bool `mapstructure:"enabled"`
+
+	// WaitForResult determines if incoming requests are blocked until the request is processed or not.
+	// Currently, this option is not available when persistent queue is configured using the storage configuration.
+	WaitForResult bool `mapstructure:"wait_for_result"`
+
+	// Sizer configuration determines the type of size measurement used by this component.
+	// It accepts "requests", "items", or "bytes".
+	Sizer exporterbatcher.SizerType `mapstructure:"sizer"`
+
+	// QueueSize represents the maximum data size allowed for concurrent storage and processing.
+	QueueSize int `mapstructure:"queue_size"`
+
+	// Blocking configuration dictates the behavior when the component's TotalSize limit is reached.
+	// If blocking is enabled, the component will wait for space; otherwise, operations will immediately return a retryable error.
+	Blocking bool `mapstructure:"blocking"`
```

Comment:
If we have `wait_for_result`, should this be called something like `block_on_overflow`? I think `blocking` might be confused the the former one.

Reply:
I like `block_on_overflow`, will do that.